### PR TITLE
manage correct shipping address invoicing a picking

### PR DESCRIPTION
**Pull request by scigghia**
add stock_transport_multi_address dependency
imp _get_invoice_vals to manage correct shipping address

fix for https://github.com/OCA/account-invoicing/issues/239

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Instead of changing dependencies in a stable version, you should check it dynamically if the field is present (something like `if <field_name> in self.env[<model>]._fields:`)

**Comment by scigghia** [comment]
thank you @pedrobaeza

File: account_invoice_shipping_address/stock.py [hunk 1]
```diff
@@ -19,16 +19,23 @@
...
+from openerp import models
 
 
-class StockPicking(orm.Model):
+class StockPicking(models.Model):
     _inherit = "stock.picking"
 
     def _get_invoice_vals(self, cr, uid, key, inv_type,
-                          journal_id, picking, context=None):
+                          journal_id, move, context=None):
```
Comment (pedrobaeza) [inline account_invoice_shipping_address/stock.py]: Why not changing also this to new API with `@api.model`? And was the signature incorrect previously?

File: account_invoice_shipping_address/stock.py [hunk 2]
```diff
@@ -19,16 +19,23 @@
...
         invoice_vals = super(StockPicking, self)._get_invoice_vals(
-            cr, uid, key, inv_type, journal_id, picking, context=context)
-        if picking and picking.partner_id:
-            invoice_vals['address_shipping_id'] = picking.partner_id.id
+            cr, uid, key, inv_type, journal_id, move, context=context)
+        if move and move.partner_id:
+            if 'delivery_address_id' in self.env['stock.picking']._fields:
+                invoice_vals['address_shipping_id'] = (
+                    move.picking_id.delivery_address_id and
```
Comment (pedrobaeza) [inline account_invoice_shipping_address/stock.py]: You can use directly `move.picking_id.delivery_address_id.id` without the fear of errors.

File: account_invoice_shipping_address/stock.py (hunk 1 above)
Comment (scigghia) [reply account_invoice_shipping_address/stock.py]: signature was incorrect 
ok for new api

File: account_invoice_shipping_address/stock.py (hunk 2 above)
Comment (scigghia) [reply account_invoice_shipping_address/stock.py]: perfect

File: account_invoice_shipping_address/stock.py [hunk 3]
```diff
@@ -19,16 +19,21 @@
...
-                          journal_id, picking, context=None):
+    @api.model
+    def _get_invoice_vals(self, key, inv_type, journal_id, move):
         invoice_vals = super(StockPicking, self)._get_invoice_vals(
-            cr, uid, key, inv_type, journal_id, picking, context=context)
-        if picking and picking.partner_id:
-            invoice_vals['address_shipping_id'] = picking.partner_id.id
+            key, inv_type, journal_id, move)
+        if move and move.partner_id:
+            if 'delivery_address_id' in self.pool['stock.picking']._fields:
```
Comment (pedrobaeza) [inline account_invoice_shipping_address/stock.py]: s/pool/env

Comment (pedrobaeza) [reply account_invoice_shipping_address/stock.py]: Put also a comment saying this field is provided by the module x.

File: account_invoice_shipping_address/stock.py [hunk 4]
```diff
@@ -30,7 +30,8 @@ def _get_invoice_vals(self, key, inv_type, journal_id, move):
         invoice_vals = super(StockPicking, self)._get_invoice_vals(
             key, inv_type, journal_id, move)
         if move and move.partner_id:
-            if 'delivery_address_id' in self.pool['stock.picking']._fields:
+            # this field is provided by stock_transport_multi_address module
+            if 'delivery_address_id' in self.pool.env['stock.picking']._fields:
```
Comment (pedrobaeza) [inline account_invoice_shipping_address/stock.py]: It's `self.env[...`, not `self.pool.env[...`

Comment (pedrobaeza) [reply account_invoice_shipping_address/stock.py]: `s/old/new` means to replace `old` by `new`

**Pull request closed without merging**